block/base/abci.go

```diff
@@ -122,12 +122,20 @@ func (l *BaseLane) ProcessLane(
 	return next(ctx, proposal)
 }
 
-// AnteVerifyTx verifies that the transaction is valid respecting the ante verification logic of
+// VerifyTx verifies that the transaction is valid respecting the ante verification logic of
 // of the antehandler chain.
-func (l *BaseLane) AnteVerifyTx(ctx sdk.Context, tx sdk.Tx, simulate bool) (sdk.Context, error) {
+func (l *BaseLane) VerifyTx(ctx sdk.Context, tx sdk.Tx, simulate bool) error {
```

Comment:
Is this necessary? Notice, every time we cache the ctx we add a significant over-head to memory consumption (this  is one of the most costly things to do in SDK). Should the caching logic not be handled by the caller?

Reply:
If we do intend to keep this logic here, maybe make note exactly what the side-effect of the fn call is

Reply:
Eh, probably not entirely necessary but if you think about execution from a deliver tx perspective, the entire context is cached before the tx is executed and state is only modified if the tx passes the expected checks.

Reply:
I think the atomicity of checking a tx is a nice feature, ill add a note that this is the behavior they should expect. In the case where they dont want this, its a super easy change